src/instructlab/model/list.py

```diff
@@ -0,0 +1,138 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+import logging
+import os
+import time
+
+# Third Party
+import click
+import pandas
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf, is_model_safetensors
+
+logger = logging.getLogger(__name__)
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--model-dirs",
+    help="Base directories where models are stored.",
+    multiple=True,
+    default=[DEFAULTS.MODELS_DIR],
+    show_default=True,
+)
+@click.option(
+    "--list-checkpoints",
+    is_flag=True,
+)
+def model_list(model_dirs: list[str], list_checkpoints: bool):
+    """lists models"""
+    data = []
+    for directory in model_dirs:
+        for entry in Path(directory).iterdir():
+            # if file, just tally the size. This must be a GGUF.
+            if entry.is_file() and is_model_gguf(entry):
+                model, age, size = _analyze_gguf(entry)
+                data.append([model, age, size])
+            elif entry.is_dir():
+                for m in _analyze_dir(entry, list_checkpoints, directory):
+                    data.append(m)
+    print_table(["Model Name", "Model Age", "Model Size"], data)
+
+
+# convert_bytes_to_proper_mag takes a dir/file size in Bytes and converts it to the proper Magnitude (MB, GB)
+def convert_bytes_to_proper_mag(f_size: float):
+    magnitudes = ["KB", "MB", "GB"]
+    magnitude = "B"
+    for mag in magnitudes:
+        if f_size >= 1024:
+            magnitude = mag
+            f_size /= 1024
+        else:
+            return f_size, magnitude
+    return f_size, magnitude
+
+
+# print_table displays the models on the system in a nicely formatted table
+def print_table(headers: list[str], data: list[list[str]]):
+    column_widths = [
+        max(len(str(row[i])) for row in data + [headers]) for i in range(len(headers))
+    ]
+    for i, header in enumerate(headers):
+        print(f"{header:{column_widths[i]}}", end=" | ")
+    print()
+
+    # Print separator line between headers and data
+    print("-" * (sum(column_widths) + 3 * len(headers) - 1))
+
+    for row in data:
+        for i, item in enumerate(row):
+            print(f"{item:{column_widths[i]}}", end=" | ")
+        print()
+
+
+AnalyzeResultGGUF = list[str]
+AnalyzeResultDir = list[list[str]]
+
+
+def _analyze_gguf(entry: Path) -> AnalyzeResultGGUF:
+    # stat the gguf, add it to the table
+    stat = Path(entry.absolute()).stat(follow_symlinks=False)
+    f_size = stat.st_size
+    f_size, magnitude = convert_bytes_to_proper_mag(f_size)
+    # add to table
+    age = pandas.to_timedelta(time.time_ns() - stat.st_mtime_ns)
+    return [entry.name, str(age), f"{f_size} {magnitude}"]
```

Comment:
```suggestion
    return [entry.name, str(age), f"{f_size:.1f} {magnitude}"]
```

You don't need full precision, printing `4.1 GiB` is more than enough.